src/client/input.ts

```diff
@@ -987,4 +1006,189 @@ export class Input extends Lister {
     }
     return retryUpload;
   }
+
+  bulkUpload({
+    inputs,
+    batchSize: providedBatchSize = 128,
+  }: {
+    inputs: GrpcInput[];
+    batchSize?: number;
+  }): Promise<void> {
+    const batchSize = Math.min(128, providedBatchSize);
+    const chunkedInputs = chunk(inputs, batchSize);
+
+    const progressBar = new cliProgress.SingleBar(
```

Comment:
I don't know if we should have CLI elements added to the SDK by default. If we want to support CLIs, or other kinds of UIs, we should have an interface by which the user passes in an EventListener object, and it receives `progress` events telling it how far along the process is. Then, users can decide whether to display a CLI widget, an HTML/Javascript component, post a line to a log file, etc.